lib/mongo/crypt/context/context.rb

```diff
@@ -42,6 +42,43 @@ def close
         true
       end
 
+      # Returns the state of the mongocrypt_ctx_t
+      #
+      # @return [ Symbol ] The context state
+      def state
+        Binding.mongocrypt_ctx_state(@ctx)
+      end
+
+      # Runs the mongocrypt_ctx_t state machine and handles
+      # all I/O on behalf of libmongocrypt
+      #
+      # @return [ String|nil ] A BSON string representing the outcome
+      #   of the state machine. This string could represent different
+      #   values depending on how the context was initialized.
```

Comment:
This method also raises on `:error`, which I think should be documented in the docstring.